.github/workflows/publish.yaml

```diff
@@ -15,6 +15,23 @@ jobs:
   build:
     runs-on: ubuntu-latest
     steps:
+
+      # Validate that the Image is published with a tag that is a valid SemVer, required by Helm chart
+      - name: Validate SemVer2 version compliance
+        if: startsWith(github.ref, 'refs/tags/')
+        env:
+          SEMVER_REGEX: ^v?(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)\.(0|[1-9][0-9]*)(-[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?(\+[0-9A-Za-z-]+(\.[0-9A-Za-z-]+)*)?$
+
+        run: |
+          ref="${{ github.ref_name }}"
```

Comment:
Having read [this](https://blog.yossarian.net/2024/12/06/zizmor-ultralytics-injection), can we avoid using bash for this check? Based on [this](https://docs.gomplate.ca/functions/regexp/), would something like

```
if: github.ref_name | RegExp.Match env.SEMVER_REGEX
```
work if it was added to the publish helm job?

Reply:
Looks good to me, I just took from the blueapi action, we can update that too after this.